disttask/framework/dispatcher/dispatcher.go

```diff
@@ -206,8 +220,66 @@ func (d *dispatcher) handleRunning() error {
 		logutil.Logger(d.logCtx).Info("previous stage finished, generate dist plan", zap.Int64("stage", d.task.Step))
 		return d.processNormalFlow()
 	}
+	// Check if any node are down.
+	handle := GetTaskFlowHandle(d.task.Type)
+	if len(d.taskNodes) == 0 {
+		schedulerIDs, err := d.taskMgr.GetSchedulerIDsByTaskID(d.task.ID)
+		if err != nil {
+			return err
+		}
+		d.taskNodes = schedulerIDs
+	}
+	d.liveNodeCnt++
+	if d.liveNodeCnt == d.liveNodeIntv {
+		d.liveNodeCnt = 0
+		serverInfos, err := GenerateSchedulerNodes(d.ctx)
+		if err != nil {
+			return err
+		}
+		eligibleServerInfos, err := handle.GetEligibleInstances(d.ctx, d.task)
+		if err != nil {
+			return err
+		}
+		newInfos := serverInfos[:0]
+		for _, m := range serverInfos {
+			found := false
+			for _, n := range eligibleServerInfos {
+				if m.ID == n.ID {
+					found = true
+					break
+				}
+			}
+			if found {
+				newInfos = append(newInfos, m)
+			}
+		}
+		d.liveNodes = newInfos
+	}
```

Comment:
Extract it into a function maybe better.
Since we want to use these functionality in rollback process too.